health: TestHealthMetric failures on release branch

A bunch of release-branch/1.78 jobs are failing with 

https://github.com/tailscale/tailscale/actions/runs/12189850751/job/34005813830?pr=14300

```
=== RUN   TestHealthMetric
=== RUN   TestHealthMetric/base-warnings
    health_test.go:389: metric value: "1", want: "2"
--- FAIL: TestHealthMetric/base-warnings (0.00s)
=== RUN   TestHealthMetric/update-warning
    health_test.go:389: metric value: "2", want: "3"
--- FAIL: TestHealthMetric/update-warning (0.00s)
--- FAIL: TestHealthMetric (0.00s)
FAIL	tailscale.com/health	1.016s
```

--- a/health/health_test.go
+++ b/health/health_test.go
@@ -14,6 +14,7 @@ import (
 	"tailscale.com/tailcfg"
 	"tailscale.com/types/opt"
 	"tailscale.com/util/usermetric"
+	"tailscale.com/version"
 )
 
 func TestAppendWarnableDebugFlags(t *testing.T) {
@@ -352,6 +353,11 @@ func TestShowUpdateWarnable(t *testing.T) {
 }
 
 func TestHealthMetric(t *testing.T) {
+	unstableBuildWarning := 0
+	if version.IsUnstableBuild() {
+		unstableBuildWarning = 1
+	}
+
 	tests := []struct {
 		desc            string
 		check           bool
@@ -361,20 +367,20 @@ func TestHealthMetric(t *testing.T) {
 	}{
 		// When running in dev, and not initialising the client, there will be two warnings
 		// by default:
-		// - is-using-unstable-version
+		// - is-using-unstable-version (except on the release branch)
 		// - wantrunning-false
 		{
 			desc:            "base-warnings",
 			check:           true,
 			cv:              nil,
-			wantMetricCount: 2,
+			wantMetricCount: unstableBuildWarning + 1,
 		},
 		// with: update-available
 		{
 			desc:            "update-warning",
 			check:           true,
 			cv:              &tailcfg.ClientVersion{RunningLatest: false, LatestVersion: "1.2.3"},
-			wantMetricCount: 3,
+			wantMetricCount: unstableBuildWarning + 2,
 		},
 	}
 	for _, tt := range tests {

